File: LeetCode/LeetCode/StackBasedOnQueue.cpp
#include <stack>
#include <queue>

using namespace std;

class MyStack
{
private:
	queue<int> queue;

public:
    MyStack() {

    }

    void push(int x) {
		queue.push(x);
    }

    int pop() {
        for (int i = 0; i < queue.size() - 1; i++)
        {
            int temp = queue.front();
			queue.pop();
			queue.push(temp);
        }
		int temp = queue.front();
		queue.pop();
		return temp;
    }

    int top() {
        int temp = this->pop();
		queue.push(temp);
		return temp;
    }

    bool empty() {
        return queue.empty();
    }
};

File: LeetCode_Csharp/LeetCode_Csharp/Code/StackAndQueue/StackBasedOnQueue.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace LeetCode_Csharp.Code.StackAndQueue
{
    public class StackBasedOnQueue
    {
        public class MyStack
        {
            private Queue<int> queue;
            public MyStack()
            {
                queue = new Queue<int>();
            }

            public void Push(int x) {
                queue.Enqueue(x);
            }
            
            public int Pop() {
                if(queue.Count == 0) return default(int);
                for(int i = 0; i < queue.Count - 1; i++) 
                {
                    int temp = queue.Dequeue();
                    queue.Enqueue(temp);
                }
                return  queue.Dequeue();
                
            }
            
            public int Top() {
                int temp = this.Pop();
                Push(temp);
                return temp;
            }
            
            public bool Empty() {
                return queue.Count == 0;
            }
        }
    }
}
